jgiven-core/src/main/java/com/tngtech/jgiven/impl/inject/ValueInjector.java

```diff
@@ -90,6 +90,8 @@ public void updateValues( Object object ) {
                 if( value != null ) {
                     field.set( object, value );
                     log.debug( "Setting field {} to value {}", field, value );
+                } else if( isRequired( field ) ) {
```

Comment:
I wonder whether the code can be optimized so that it does not have to iterate twice over the annotations of the field. Because the getResolution method also does the iteration.